skbio/alignment/tests/test_tabular_msa.py

```diff
@@ -493,6 +493,43 @@ def test_reindex_non_hashable_keys(self):
 
         npt.assert_array_equal(msa.keys, keys)
 
+    def test_append(self):
+        msa = TabularMSA([DNA('ACGT'), DNA('TGCA')])
+        msa.append(DNA('AAAA'))
+        self.assertEqual(msa.shape, (3, 4))
+        # TODO: Hack to get last seq. once __getitem__ is implemented use
+        # msa[-1]
+        seq = None
+        for seq in msa:
+            pass
+        self.assertEqual(seq, DNA('AAAA'))
+
+    def test_append_to_empty_msa(self):
+        msa = TabularMSA([])
+        msa.append(DNA('ACGT'))
+        self.assertEqual(len(msa), 1)
+
+
+    def test_append_to_empty_msa_invalid_dtype(self):
```

Comment:
For each of these error-checking tests, can you assert that the object's state isn't altered? This will catch the case where `self._seqs` is appended to before an error is raised, which would put the object into an inconsistent state. This is important to avoid, for example, if the user catches the error and continues to interact with the MSA. There are some examples in the other `TabularMSA` unit tests for how we're checking this.